### DISCUSSION TITLE
Mini Synth version (packaging only a very small amount of modules)

### DISCUSSION THREAD

**Discussion by falkTX** [DISTRHO/Cardinal#398] [Ideas]
Was suggested on IRC, I am thinking it makes a lot of sense.

We can have a minified Cardinal build that includes only the essentials for modular synth creation, and give these modules extra attention (as in, testing, polishing, fixing etc)
Also makes it easier to approach for the not-sure-how-to-do-it-yet Remote control feature, by only needing to care about a very few select modules.

Open to opinions and suggestions on which modules to include.
Fundamental and Bogaudio have quite the nice starting point, but we do not necessarily need their entire module collection.
Personally I am against any module that loads samples to be in this minified version.
Also, FX can be skipped, including Carla and Ildaeil.

So let's pick at least 1, maximum 3, of the best modules for each category.
Would be great if all modules are polyphonic, so we dont have to pick and choose for the feature (if something is really nice but mono, we can consider patching its sources to make it poly).

Modules to include follow, categories in alphabetical order.
The ones without [x] mean undecided

### Host Tools
- [x] Host Audio 2
- [x] Host CV (only 5 ports instead of 10)
- [x] Host MIDI
- [x] Host MIDI CC
- [x] Host MIDI Gate
- [x] Host MIDI Map
- [x] Host Params
- [x] Host Params Map
- [x] Host Time

### Envelope
- [ ] Mindmeld Shapemaster
- [x] VCV ADSR

### LFO
- [x] Bogaudio LFO
- [x] VCV LFO
- [ ] VCV WT LFO (loads files)

### Filter
- [ ] Autinn Bass
- [x] Bogaudio VCF
- [x] VCV VCF

### Mixer
- [ ] Bogaudio MATRIX18
- [ ] Bogaudio XFADE
- [x] VCV VCA Mix

### Noise
- [x] Bogaudio Noise
- [x] VCV Noise
- [x] VCV Random

###  Oscillator
- [x] Audible instrument Macro Osc 2
- [x] Bogaudio VCO
- [x] Valley Dexter
- [x] Valley Interzone

### Sample & Hold
- [x] Bogaudio Sample & Hold

### Sampler
- [ ] LOMAS SAMPLER (custom interaction)

### Sequencer
- [ ] ARP: Hampton Harmonics ARP
- [ ] Bogaudio ADDR-SEQ +ASX
- [ ] Eugene (Euclidian sequencer)
- [ ] Kocmoc TRG

### Utilities
- [x] Aria Spleet
- [x] Aria Swerge
- [ ] BaconPlugs Glissinator
- [ ] Bidoo lATe
- [x] Bogaudio Offset
- [x] Bogaudio Switch
- [x] Bogaudio Switch18
- [ ] Bogaudio UMix (removed, other modules do it better)
- [ ] Bogaudio ''+Logic modules"
- [ ] Bogaudio vu meter...
- [ ] Cardinal Audio to CV Pitch (needs fftw, analyzer display will never be correct)
- [ ] cf label (but GLUE would be better)
- [ ] Impromptu Clocked
- [ ] Mockba Modular Countah
- [ ] Repelzen Re-Win Quantizer
- [x] Text Editor
- [x] VCV Merge
- [x] VCV Quantizer
- [x] VCV Scope
- [x] VCV Split

**Comment by subderground** [comment]
It's difficult to chose but here is a list :-)
LFO: 
- BOGAUDIO LFO
- VCV WT LFO
- VCV Random (as S&H section too) 

ENV:
VCV ADSR
Mindfield Shapemaster (as LFO too)

MIXER:
BOGAUDIO XFADE
BOGAUDIO MATRIX18
VCV VCA Mix

NOISE:
BOGAUDIO NOISE
VCV NOIZ

SEQ:
BOGAUDIO ADDR-SEQ +ASX,
Kocmoc TRG
ARP: Hampton Harmonics ARP

VCA:
VCV VCA

VCF:
BOGAUDIO VCF
VCV VCF
Autinn Bass
(I haven't found THE Filter module yet.)

VCO:
BOGAUDIO VCO
Audible instrument Macro Osc 2
Valley Dexter

SAMPLER (starts without any sample inside):
LOMAS SAMPLER

UTILS:
BOGAUDIO UMIX, SWITCH, SWITCH18, OFFSET, ''+Logic modules", vu meter... (Hum...Bogaudio modules are great !)
VCV scope
BACON Glissinator
Impromptu Clocked
Bidoo lATe
Mockba Modular Countah
Repelzen Re-Win Quantizer
cf label (but GLUE would be better)
(...)
Have a nice day.

**Reply by subderground** [reply]
(EDIT°) And...
Rare Breeds - Eugene (Euclidian sequencer)
S&H = BGA S&H

**Comment by alanfrazier** [comment]
I use the MIDI CC module quite a bit.  And I'm not sure I've had a patch yet that didn't have Plateau in it.  I don't know if that is enough to justify their inclusion, though.

Oh, and I use the default VCV ones most of the time (because they're polyphonic).

**Reply by dromer** [reply]
midi can be considered "internal modules" so I suspect all of those will be present anyway.
Plateau may be too heavy for such a mini version.

Also wondering about polyphony preference in general. Should we have at least one polyphonic version per category?

**Reply by alanfrazier** [reply]
I just noticed that falkTX didn't have the MIDI CC one checked in his list.

**Reply by alanfrazier** [reply]
If the purpose of this configuration is for testing, though, then my input is irrelevant.  I was just thinking about what I'd want for a minimal setup in an embedded system, and the default VCV and Plateau would cover it nicely for me (and that could very well be only because I'm a super-n00b at modular synthesis).

**Reply by dromer** [reply]
Yeah I think all midi objects would be nice. CC certainly useful.

**Comment by seastones** [comment]
If you're considering something like this, would it be possible to work on a configurable list of plugins to compile in general? (Not discounting that it would be a large undertaking, mind. Just possible. Ideally it could also streamline adding new modules?)

**Reply by falkTX** [reply]
no configurable plugins, that would be way too much setup.
I am just going with a good set of modules and very harsh/limiting set of rules.

PS: This is going for the "running a remote instance" case too, so anything that has custom animations or meters cannot go in.

**Reply by seastones** [reply]
I figured it would be a nightmare to set up and possibly maintain, but more it was a question of "can that be done?" than a suggestion. (With the idea that modules that can't be included compiled being able to be compiled by the end user in a similar way as FFMPEG.)

Thank you for the prompt reply!

**Comment by korg-supporter13** [comment]
I don't think VCV Noise is Polyphonic.
